client/src/templates/Challenges/quiz/show.tsx

```diff
@@ -111,147 +159,134 @@ class ShowQuiz extends Component<ShowQuizProps, ShowQuizState> {
       helpCategory
     });
     challengeMounted(challengeMeta.id);
-    this.container.current?.focus();
-  }
+    container.current?.focus();
+    // This effect should be run once on mount
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
 
-  componentDidUpdate(prevProps: ShowQuizProps): void {
-    const {
-      data: {
-        challengeNode: {
-          challenge: { title: prevTitle }
-        }
-      }
-    } = prevProps;
-    const {
-      challengeMounted,
-      data: {
-        challengeNode: {
-          challenge: { title: currentTitle, challengeType, helpCategory }
-        }
-      },
-      pageContext: { challengeMeta },
-      updateChallengeMeta
-    } = this.props;
-    if (prevTitle !== currentTitle) {
-      updateChallengeMeta({
-        ...challengeMeta,
-        title: currentTitle,
-        challengeType,
-        helpCategory
-      });
-      challengeMounted(challengeMeta.id);
+  useEffect(() => {
+    updateChallengeMeta({
+      ...challengeMeta,
+      title,
+      challengeType,
+      helpCategory
+    });
+    challengeMounted(challengeMeta.id);
+  }, [
+    title,
+    challengeMeta,
+    challengeType,
+    helpCategory,
+    challengeMounted,
+    updateChallengeMeta
+  ]);
+
+  const handleAnswersCheck = () => {
+    const unansweredQuestions = quizData.reduce<number[]>(
+      (acc, curr, id) => (curr.selectedAnswer == null ? [...acc, id + 1] : acc),
+      []
+    );
+
+    if (unansweredQuestions.length) {
+      setErrorMessage(
+        t('learn.quiz.unanswered-questions', {
+          unansweredQuestions: unansweredQuestions.join(', ')
+        })
+      );
+      return;
     }
-  }
 
-  handleSubmit() {
-    console.log('handleSubmit');
-  }
+    const correctCount = quizData.reduce(
+      (acc, curr) => (curr.selectedAnswer === curr.correctAnswer ? ++acc : acc),
+      0
+    );
```

Comment:
There's another way: `errorMessage` doesn't have to be state, since its value can be derived from `hasSubmitted`, `quizData`, `correctAnswerCount` and `quiz`. i.e. state and props that already exist.

That means `errorMessage` is "redundant state" which is something of anti-pattern.

Since changing this touches a bunch of things, I'll just push a commit so you can see what I'm talking about.